chain/ethereum/src/adapter.rs

```diff
@@ -433,6 +495,13 @@ pub(crate) struct EthereumBlockFilter {
 }
 
 impl EthereumBlockFilter {
+    pub fn from_mapping(mapping: &Mapping) -> Self {
+        Self {
+            contract_addresses: HashSet::new(),
+            trigger_every_block: mapping.block_handlers.len() != 0,
+        }
```

Comment:
This isn't taking into account call filters on blocks https://thegraph.com/docs/en/developer/create-subgraph-hosted/#supported-filters

Reply:
Is this available for templates as well?

Reply:
It is afaik, you can test it out to be sure

Reply:
I think we need to have this broader filter for now in this version, we can optimise it later. 

Making this change is going to require wider changes, the Block filter properties are used directly instead of using a match statement and the firehose filters don't support this either